packages/odyssey-react-theme/src/withTheme/stylesheet.ts

```diff
@@ -17,6 +17,8 @@ export class OStyleSheet extends Set<string> {
     super();
     const el = document.createElement("style");
     el.setAttribute("data-ods", "");
+    // @ts-expect-error ts(2339)
+    el.setAttribute("nonce", window.nonce);
```

Comment:
In the admin app, the global variable is `window.cspNonce`. I don't we should hardcode this if we can avoid it since odyssey is a library. Can we make it a configuration variable instead so that each repo which uses odyssey can manage their own nonce and pass it to odyssey?

Reply:
Are we using different variable names in different apps, or it's all `window.cspNonce`?

Reply:
This could definitely be configurable. I like that idea better. But instead, we should have apps pass us the nonce, not infer it from `window`.

The problem is this file `factory` which news up our object. Even if we wanted to configure it, we can't without a significant rewrite:
https://github.com/okta/odyssey/blob/64f6827d1430da67f69270fbf2eb3ed8cb867688/packages/odyssey-react-theme/src/withTheme/factory.tsx#L24

Another idea is exporting the new'd up `style` instance and adding a configuration setting to add that nonce, but by that point, it may be too late. We really need to add this in the `constructor`.

Reply:
I know admin and super user are using cspNonce. I don't know if anything else is using odyssey. If odyssey is only gonna be an internal library I guess it's NBD to keep it as global variable. Can you manually test it before merging to make sure the warning goes way?

Reply:
It's an internal library, but it's also open source. That particular Odyssey is deprecated, so I think it's not a big deal at this point.